Bottom: c87d40f9917c21534131c2f4ac21976842d261b2
Top:    2001722b372393356bf5e5d9bb08cf736e1577eb
Author: Stanislav Brabec <sbrabec@suse.cz>
Date:   2011-07-18 17:39:52 +0200

RipUp(): Rip arcs created by toporouter.

Toporouter creates not only lines but also arcs.
Rip up all auto routed tracks should rip these arcs as well.

Closes-bug: lp-812380

Reviewed-by: Peter Clifton <pcjc2@cam.ac.uk>


---

diff --git a/src/action.c b/src/action.c
index 9b67924..d6e1d89 100644
--- a/src/action.c
+++ b/src/action.c
@@ -3683,6 +3683,15 @@ ActionRipUp (int argc, char **argv, int x, int y)
 	      }
 	  }
 	  ENDALL_LOOP;
+	  ALLARC_LOOP (PCB->Data);
+	  {
+	    if (TEST_FLAG (AUTOFLAG, arc) && !TEST_FLAG (LOCKFLAG, arc))
+	      {
+		RemoveObject (ARC_TYPE, layer, arc, arc);
+		changed = true;
+	      }
+	  }
+	  ENDALL_LOOP;
 	  VIA_LOOP (PCB->Data);
 	  {
 	    if (TEST_FLAG (AUTOFLAG, via) && !TEST_FLAG (LOCKFLAG, via))
